src/Sylius/Bundle/MediaBundle/EventListener/ImageMediaReferenceListener.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\MediaBundle\EventListener;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Sylius\Component\Media\Model\ImageInterface;
+use Symfony\Cmf\Bundle\MediaBundle\ImageInterface as CmfImageInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
+ */
+class ImageMediaReferenceListener implements EventSubscriber
+{
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(
+        ContainerInterface $container
```

Comment:
Because this is class a Doctrine event subscriber we're going to have a `ServiceCircularReferenceException` .

I've injected RegistryManager service whicha  better choice than container ;) :+1: